evolver/app/tests/test_calibration.py

```diff
@@ -72,196 +67,102 @@ def test_temperature_calibration_procedure_actions():
 
-    # Dispatch the action to calculate the fit
-    fit_action_payload = {"action_name": "calculate_vial_0_fit"}
-    fit_dispatch_response = client.post("/hardware/test/calibrator/procedure/dispatch", json=fit_action_payload)
+    fit_dispatch_response = dispatch_action(client, "test", "calculate_vial_0_fit")
     assert fit_dispatch_response.status_code == 200
-    assert fit_dispatch_response.json() == {"0": {"reference": [25.0], "raw": [1.23]}}
 
     output_transformer_response = client.get("/hardware/test/calibrator/output_transformer")
     assert output_transformer_response.status_code == 200
     assert output_transformer_response.json()["0"]["parameters"] == [0.6149999999999997, 0.024599999999999997]
 
 
 def test_get_calibration_data():
-    # Set up the evolver instance with hardware and a Temperature Calibrator
-    temp_calibrator = TemperatureCalibrator(
-        input_transformer={
-            0: LinearTransformer(),
-            1: LinearTransformer(),
-            2: LinearTransformer(),
-        },
-        output_transformer={
-            0: LinearTransformer(),
-            1: LinearTransformer(),
-            2: LinearTransformer(),
-        },
-    )
-
-    # Create NoOpSensorDriver and assign the temp calibrator to it
-    evolver_instance = Evolver(
-        hardware={"test": NoOpSensorDriver(name="test", calibrator=temp_calibrator, vials=[0, 1, 2])}
-    )
-    temp_calibrator.evolver = evolver_instance
-    app.state.evolver = evolver_instance
-
-    # Mock the hardware read method
-    evolver_instance.hardware["test"].read = lambda: [1.23, 2.34, 3.45]
-    client = TestClient(app)
+    temp_calibrator, client = setup_evolver_with_calibrator(TemperatureCalibrator)
 
-    # Initialize the calibration procedure
-    calibration_data_response = client.post("/hardware/test/calibrator/procedure/start")
-    assert calibration_data_response.status_code == 200
+    app.state.evolver.hardware["test"].read = lambda: [1.23, 2.34, 3.45]
 
-    # Dispatch actions to set reference value, read raw value, and calculate fit
-    reference_dispatch_response = client.post(
-        "/hardware/test/calibrator/procedure/dispatch",
-        json={"action_name": "measure_vial_0_temperature", "payload": {"temperature": 25.0}},
-    )
-    assert reference_dispatch_response.status_code == 200
-    assert reference_dispatch_response.json()["0"]["reference"] == [25.0]
+    client.post("/hardware/test/calibrator/procedure/start")
 
-    raw_dispatch_response = client.post(
-        "/hardware/test/calibrator/procedure/dispatch", json={"action_name": "read_vial_0_raw_output", "payload": {}}
-    )
-    assert raw_dispatch_response.status_code == 200
-    assert raw_dispatch_response.json()["0"]["raw"] == [1.23]
+    dispatch_action(client, "test", "measure_vial_0_temperature", {"temperature": 25.0})
+    dispatch_action(client, "test", "read_vial_0_raw_output")
+    dispatch_action(client, "test", "calculate_vial_0_fit")
```

Comment:
I think this is fine for now, but we might want to make a note to create a mock procedure for the tests which cover action kinds (e.g. ones that need input, ones that don't, etc.), iso using the temperature one - just because this looks more generic